### PR TITLE
Save point in database

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -30,4 +30,4 @@ src/main/webapp/snoop.jsp
 *.ear
 # virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
 hs_err_pid*
-/bin/
+/bin/
```
Comment (Sabfir) [inline .gitignore]: revert this changes, plz

File: src/main/java/ua/softserve/rv_018/greentourism/model/Place.java [hunk 2]
```diff
@@ -38,6 +38,14 @@
 	
 	public Place() { }
 	
+	public Place(int id, String name, String description) {
```
Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/model/Place.java]: where do you use this constructor?

File: src/main/java/ua/softserve/rv_018/greentourism/repository/PlaceRepository.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package ua.softserve.rv_018.greentourism.repository;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import ua.softserve.rv_018.greentourism.model.Place;
+
+public interface PlaceRepository extends JpaRepository<Place, Integer> {
+
+}
```
Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/repository/PlaceRepository.java:27]: Put this in one line:
public interface PlaceRepository extends JpaRepository<Place, Integer> {}

File: src/main/webapp/resources/app/components/addpoint/addpoint.component.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
+'use strict';
```
Comment (KostyaTk) [inline src/main/webapp/resources/app/components/addpoint/addpoint.component.js]: Pls change code format in this file. It's hard to read it.

**Comment by ghost** [comment]
Thank's I'm working on it . Also i know about another problems, and i want to resolve it. After that i will do new commit in this pull request.

**Comment by ghost** [comment]
I merged dev in my branch, and working on all mistake that I have. if everything fine, I'll do new commit.

File: src/main/java/ua/softserve/rv_018/greentourism/controller/PlaceController.java [hunk 5]
```diff
@@ -79,4 +81,20 @@
 				.findByNameIgnoreCaseContaining(name);
 		return new ResponseEntity<>(places, HttpStatus.OK);
 	}
+	
+	@RequestMapping(value = "/addplace", method = RequestMethod.POST, headers = "Accept=application/json", produces = {
```
Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/controller/PlaceController.java]: Here should be url "/place".

File: src/main/java/ua/softserve/rv_018/greentourism/model/Place.java [hunk 6]
```diff
@@ -96,4 +96,4 @@ public String toString() {
 				+ ", owner id = " + user.getId()
 				+ ", category = " + category.getName() + "]";
 	}
-}
+}
```
Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/model/Place.java]: You just deleted last line, plz revert changes

File: src/main/java/ua/softserve/rv_018/greentourism/repository/PlaceRepository.java [hunk 7]
```diff
@@ -10,3 +10,4 @@
 public interface PlaceRepository extends JpaRepository<Place, Long> {
 	List<Place> findByNameIgnoreCaseContaining(String name);
 }
+
```
Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/repository/PlaceRepository.java]: plz, revert changes

File: src/main/java/ua/softserve/rv_018/greentourism/service/PlaceService.java [hunk 8]
```diff
@@ -10,4 +10,5 @@
      */
     Collection<Place> findAll();
     Collection<Place> findByNameIgnoreCaseContaining(String name);
+    Place create(Place place);
```
Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/service/PlaceService.java:32]: plz, put empty line between methods (java code convention)

File: src/main/java/ua/softserve/rv_018/greentourism/service/PlaceServiceImpl.java [hunk 9]
```diff
@@ -42,4 +42,10 @@
 	public Collection<Place> findByNameIgnoreCaseContaining(String name) {
 		return placeRepository.findByNameIgnoreCaseContaining(name);
 	}
+	
+	@Override
+	public Place create(Place place) {
+		Place addedPlace = placeRepository.save(place);
+		return addedPlace;
```
Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/service/PlaceServiceImpl.java]: plz, log your code

File: src/main/java/ua/softserve/rv_018/greentourism/service/PlaceService.java (hunk 8 above)
Comment (Sabfir) [inline src/main/java/ua/softserve/rv_018/greentourism/service/PlaceService.java:32]: Plz, write javadoc before method, refer to User functionality

**Comment by KostyaTk** [comment]
Reviewed

File: src/main/webapp/resources/app/components/map/map.component.js [hunk 10]
```diff
@@ -3,44 +3,70 @@
 angular.module('greenApp')
   .component('map', {
     templateUrl: _contextPath + '/resources/app/components/map/map.template.html',
-    controller: function($rootScope, $scope) {
-    	var mymap = L.map('mapid').setView([51.505, -0.09], 13);
+    controller: function($rootScope, $scope, $http) {
+    	var mymap  = L.map('mapid').setView([ 50.619900, 26.251617 ], 13);
```
Comment (Sabfir) [inline src/main/webapp/resources/app/components/map/map.component.js]: Delete plz extra space before =

File: src/main/webapp/resources/app/components/map/map.template.html [hunk 11]
```diff
@@ -1 +1,70 @@
-<div id="mapid"></div>
\ No newline at end of file
+<div class="content">
+
+	<div class="map-wrapper">
+		<div id="mapid"></div>
+	</div>
+
+
```
Comment (Sabfir) [inline src/main/webapp/resources/app/components/map/map.template.html:4]: Plz, delete extra empty line

File: src/main/webapp/resources/app/components/map/map.template.html [hunk 12]
```diff
@@ -1 +1,70 @@
-<div id="mapid"></div>
\ No newline at end of file
+<div class="content">
```
Comment (Sabfir) [inline src/main/webapp/resources/app/components/map/map.template.html]: Why are you deleted empty line here?

File: src/main/webapp/resources/app/components/map/map.template.html [hunk 13]
```diff
@@ -1 +1,70 @@
...
+
+
+	<div class="fixed-action-btn" ng-show="addButtonAddPlace" ng-hide = "false">
+		<a ng-click="addPlaceMenu(); hideButtonAddPlace()" data-activates="slide-out"
+			class="button-collapse btn-floating btn-large red modal-trigger">
+			<i class="large material-icons">mode_edit</i>
+		</a>
+	</div>
+
+
```
Comment (Sabfir) [inline src/main/webapp/resources/app/components/map/map.template.html]: Plz, delete extra empty line

File: src/main/webapp/resources/app/css/style.css [hunk 14]
```diff
@@ -0,0 +1,278 @@
...
+    font-size: 14px;
+    height: 34px;
+    line-height: 1.42857;
+    padding: 6px 12px;
+    transition: border-color 0.15s ease-in-out 0s, box-shadow 0.15s ease-in-out 0s;
+    width: 100%;
+}
+
+    
+
```
Comment (Sabfir) [inline src/main/webapp/resources/app/css/style.css]: Plz, delete extra empty lines and spaces

File: src/main/webapp/resources/app/css/style.css [hunk 15]
```diff
@@ -0,0 +1,278 @@
...
+/*-------------- Place for map --------------*/
+.map-wrapper {
+  position: absolute;
+  padding-top: 0px; // px that has our nav
+  top: 0;
+  left: 0;
+  height: 100%;
+  width: 100%;
+  z-index: 1;
+}
```
Comment (Sabfir) [inline src/main/webapp/resources/app/css/style.css]: plz, add empty line

File: src/main/webapp/resources/app/css/style.css [hunk 16]
```diff
@@ -0,0 +1,278 @@
...
+/*-------------- Red button --------------*/
+.fixed-action-btn {
+	bottom: 45px;
+	right: 24px;
+	animation: 0.4s ease-out 0s normal none 1 running expandFab;
+}
+
+
+/*-------------- ng-animate --------------*/
+
```
Comment (Sabfir) [inline src/main/webapp/resources/app/css/style.css]: You should have an unique style.
If you add empty line between comments and code, so do it everywhere, or delete it everywhere.
Plz, do control of this file for this problem

File: src/main/webapp/resources/app/css/style.css [hunk 17]
```diff
@@ -0,0 +1,278 @@
...
+a.sideNavMenuButton {
+	margin-left: 1%;
+}
+
+#sidenav-overlay {
+	background-color: rgba(0, 0, 0, 0) !important;
+}
+
+
```
Comment (Sabfir) [inline src/main/webapp/resources/app/css/style.css]: extra empty line should be deleted

File: src/main/webapp/resources/app/css/style.css [hunk 18]
```diff
@@ -0,0 +1,278 @@
...
+}
+
+.content {
+	height: 95vh;
+	width: 100%;
+	margin-left: 0;
+	margin-top: 0;
+	transform: translate3d(0, 0, 0);
+	transition: transform 0.4s ease;
+
```
Comment (Sabfir) [inline src/main/webapp/resources/app/css/style.css]: Extra empty line should be deleted

File: src/main/java/ua/softserve/rv_018/greentourism/model/Point.java [hunk 19]
```diff
@@ -8,7 +8,6 @@
 import javax.persistence.Table;
 
 import com.fasterxml.jackson.annotation.JsonIgnore;
-import com.fasterxml.jackson.annotation.JsonProperty;
```
Comment (ghost) [inline src/main/java/ua/softserve/rv_018/greentourism/model/Point.java:11]: eclipse shows that line like a unnecessary import

File: src/main/webapp/resources/app/components/map/map.template.html (hunk 12 above)
Comment (ihorkazakov) [inline src/main/webapp/resources/app/components/map/map.template.html]: there is empty line between this two block.
I just wrapped <div id="mapid"></div> in aditional block 

<div class="map-wrapper">
		<div id="mapid"></div>
</div>

Comment (Sabfir) [reply src/main/webapp/resources/app/components/map/map.template.html:1]: sry, my fault

File: src/main/java/ua/softserve/rv_018/greentourism/model/Point.java (hunk 19 above)
Comment (Sabfir) [reply src/main/java/ua/softserve/rv_018/greentourism/model/Point.java:11]: cool

**Comment by Sabfir** [comment]
Reviewed!

**Comment by 47souls** [comment]
Reviewed.

File: src/main/webapp/resources/app/components/map/map.component.js [hunk 20]
```diff
@@ -3,43 +3,68 @@
...
+				category : $scope.newPlaceType,
+				description : $scope.newPlaceDescription,
+				point : {
+					latitude : $scope.latitude,
+					longitude : $scope.longitude
+				}
+			};
+
+			$http.post(_contextPath + "/api/place/", dataObj)
+			.success(function(data, status, headers, config) {});
```
Comment (ghost) [inline src/main/webapp/resources/app/components/map/map.component.js]: then call back for true, then call back error

Comment (Michaluch) [inline src/main/webapp/resources/app/components/map/map.component.js]: Promise methods success and error have been deprecated. Use the standard then method instead

File: src/main/webapp/resources/app/components/map/map.component.js [hunk 21]
```diff
@@ -3,43 +3,68 @@
...
+		};
+		
+		$scope.resetAddPlaceForm = function(form) {
```
Comment (Michaluch) [inline src/main/webapp/resources/app/components/map/map.component.js:78]: Why you wrote this method. This can make by using HTML. Maybe have reason only for removing marker from map

File: src/main/webapp/resources/app/components/map/map.template.html [hunk 22]
```diff
@@ -1 +1,64 @@
...
+					<span class="title">1. Choose place on map:</span><br> <span
+						class="title">Latitude:</span> <input type="text" class="latlng"
+						id="latitude" ng-model="latitude" disabled="disabled"></span> <span
+						class="title">Longitude:</span> <input type="text" class="latlng"
+						id="longitude" ng-model="longitude" disabled="disabled"></span>
+				</div>
+				<div class="form-group">
+					<label for="placeName"> <span class="title">2. Enter
+							place name:</span>
+					</label> <input type="text" name="name" class="form-control"
```
Comment (Michaluch) [inline src/main/webapp/resources/app/components/map/map.template.html:28]: If this field is required you should telling for user about it, therefore add some field for small hint about this field.

**Comment by Bmaks-github** [comment]
Reviewed.

File: src/main/webapp/resources/app/components/map/map.template.html [hunk 23]
```diff
@@ -1,7 +1,69 @@
...
+						id="newPlaceName" ng-model="newPlaceName" ng-minlength="5" ng-maxlength="70" required>
+				</div>
+				<div ng-messages="placeMaker.name.$error">
+					<p ng-message="minlength">Place name is too short.</p>
```
Comment (Michaluch) [inline src/main/webapp/resources/app/components/map/map.template.html]: This message less informative. Use more ifmorming style like "Choose a longer username, at least four characters. "
